packages/playwright-core/src/server/browserContext.ts

```diff
@@ -610,11 +605,8 @@ export abstract class BrowserContext extends SdkObject {
         for (const originState of state.origins) {
           const frame = page.mainFrame();
           await frame.goto(metadata, originState.origin);
-          await frame.evaluateExpression(`
-            originState => {
-              for (const { name, value } of (originState.localStorage || []))
-                localStorage.setItem(name, value);
-            }`, { isFunction: true, world: 'utility' }, originState);
+          const args: Parameters<typeof storageScript.restore> = [originState];
+          await frame.evaluateExpression(storageScript.restore.toString(), { isFunction: true, world: 'utility' }, args);
```

Comment:
Looking at the old code, we did not need to wrap `originState` into an array, but in the new code we do. What has changed?